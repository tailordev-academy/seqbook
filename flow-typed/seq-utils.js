import type { NtSeq } from '../src/types';
import type { Sequence } from '../src/reducers/app';

declare module "seq-utils" {
  declare export function generate(): Sequence;
  declare export function createSequenceFromDNA(dna: string): NtSeq;
}
