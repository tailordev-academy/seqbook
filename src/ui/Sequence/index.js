/* @flow */
import React from 'react';

import './styles.css';

type Props = {|
  sequence: string,
|};

const Sequence = ({ sequence }: Props) => (
  <pre className="Sequence">
    <code>{sequence}</code>
  </pre>
);

export default Sequence;
