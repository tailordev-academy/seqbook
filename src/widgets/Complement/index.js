import React from 'react';
import PropTypes from 'prop-types';

import Sequence from 'ui/Sequence';

const Complement = ({ sequence }) => {
  return (
    <div className="Complement">
      <h3>Complement</h3>

      {sequence.size() === 0
        ? <p>sequence is empty</p>
        : <Sequence sequence={sequence.complement().sequence()} />}
    </div>
  );
};

Complement.propTypes = {
  sequence: PropTypes.shape({
    complement: PropTypes.func.isRequired,
  }).isRequired,
};

export default Complement;
