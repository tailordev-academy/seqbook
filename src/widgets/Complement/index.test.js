import React from 'react';
import { shallow } from 'enzyme';
import { createSequenceFromDNA } from 'seq-utils';
import renderer from 'react-test-renderer';

import Sequence from 'ui/Sequence';

import Complement from './index';

it('renders correctly', () => {
  const tree = renderer
    .create(<Complement sequence={createSequenceFromDNA('AAA')} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a message when sequence is empty', () => {
  const wrapper = shallow(<Complement sequence={createSequenceFromDNA('')} />);
  expect(wrapper.find(Sequence)).toHaveLength(0);
  expect(wrapper.text()).toContain('sequence is empty');
});

it('uses a Sequence', () => {
  const wrapper = shallow(
    <Complement sequence={createSequenceFromDNA('AAA')} />
  );
  expect(wrapper.find(Sequence)).toHaveLength(1);
});
