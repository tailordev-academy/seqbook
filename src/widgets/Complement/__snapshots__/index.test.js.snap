// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Complement"
>
  <h3>
    Complement
  </h3>
  <pre
    className="Sequence"
  >
    <code>
      TTT
    </code>
  </pre>
</div>
`;
